subcommand/acl-controller/command.go

```diff
@@ -102,27 +101,47 @@ func (c *Command) run() error {
 	// Set up ECS client.
 	ecsClient := ecs.New(clientSession)
 
-	cfg := api.DefaultConfig()
-	if caCert := os.Getenv(consulCACertEnvVar); caCert != "" {
-		cfg.TLSConfig = api.TLSConfig{
-			CAPem: []byte(caCert),
-		}
+	serverConnMgrCfg, err := c.config.ConsulServerConnMgrConfig(ecsMeta)
+	if err != nil {
+		return fmt.Errorf("constructing server connection manager config: %s", err)
+	}
+
+	watcher, err := discovery.NewWatcher(c.ctx, serverConnMgrCfg, c.log)
+	if err != nil {
+		return fmt.Errorf("unable to create consul server watcher: %s", err)
+	}
+
+	go watcher.Run()
+	defer watcher.Stop()
+
+	state, err := watcher.State()
+	if err != nil {
+		return fmt.Errorf("unable to fetch consul server watcher state: %s", err)
+	}
+
+	// Client config for the client that talks directly to the server agent
+	cfg := c.config.ClientConfig()
+	cfg.Address = net.JoinHostPort(state.Address.IP.String(), strconv.FormatInt(int64(c.config.ConsulServers.HTTPPort), 10))
+
+	token := os.Getenv(bootstrapTokenEnvVar)
+	if token != "" {
+		cfg.Token = token
 	}
 
 	consulClient, err := api.NewClient(cfg)
 	if err != nil {
-		return err
+		return fmt.Errorf("constructing consul client from config: %s", err)
```

Comment:
nit: Use `%w` for error wrapping. Also clarity suggestion ("consul client" vs "consul api client")

```suggestion
		return fmt.Errorf("constructing Consul API client from config: %w", err)
```